tests/device_random_test.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+#include <aws/common/device_random.h>
+
+#include <aws/common/byte_buf.h>
+
+#include <aws/testing/aws_test_harness.h>
+
+static int s_device_rand_u64_fn(struct aws_allocator *allocator, void *ctx) {
+    (void)allocator;
+    (void)ctx;
+
+    uint64_t last_value = 0;
+
+    for (size_t i = 0; i < 100000; ++i) {
+        uint64_t next_value = 0;
+        ASSERT_SUCCESS(aws_device_random_u64(&next_value));
+        ASSERT_FALSE(next_value == last_value);
```

Comment:
It is perfectly valid for the same number to come twice in sequence, even if astronomically unlikely. Also, you'd want to do ASSERT_NOT_EQUAL() here so we can see the values.

A better test might be to make a set and see if it is like 95% unique or something like that.

Reply:
There is no ASSERT_NOT_EQUAL at the moment, I don't have a set structure to use ( I suppose we could just use the hash table but that's a lot of work when really this is just a sanity check to make sure we're actually generating random values). Whatever test we do will be probablistic anyways so I'd prefer to keep it as is for the moment.

Reply:
Ok, fair enough